### PR TITLE
Add volume workload

### PR DESCRIPTION
**Pull request by AbdelrahmanElawady**
### Description

Add volume workload

### Changes

Add volume workload

### Related Issues

- #993 

### Checklist

- [x] Tests included
- [x] Build pass
- [x] Documentation
- [x] Code format and docstring

### CODE REVIEW

File: grid-client/workloads/deployment.go [hunk 1]
```diff
@@ -26,13 +26,18 @@ type Deployment struct {
...
 	QSFS             []QSFS
+	Volumes          []Volume
 
 	// computed
 	NodeDeploymentID map[uint32]uint64
 	ContractID       uint64
 	IPrange          string
 }
 
+// TODO: NewDeployment should take a list of Workload interface instead of defining
```
Comment (rawdaGastan) [inline grid-client/workloads/deployment.go:37]: I think we should open an issue for it

File: grid-client/workloads/deployment.go [hunk 2]
```diff
@@ -161,6 +175,10 @@ func (d *Deployment) ZosDeployment(twin uint32) (gridtypes.Deployment, error) {
 		}
 		wls = append(wls, qWls)
 	}
+	for _, v := range d.Volumes {
+		volumeWl := v.ZosWorkload()
+		wls = append(wls, volumeWl)
```
Comment (rawdaGastan) [inline grid-client/workloads/deployment.go]: should be `append(wls, v.ZosWorkload())`

File: tfrobot/pkg/deployer/deployer.go [hunk 3]
```diff
@@ -242,7 +242,7 @@ func buildDeployments(vms []Vms, nodesIDs []int, sshKeys map[string]string) grou
 			network := buildNetworkDeployment(vmGroup, nodeID, vmName, solutionType)
 			vm := buildVMDeployment(vmGroup, vmName, network.Name, sshKeys[vmGroup.SSHKey], mounts)
 
-			deployment := workloads.NewDeployment(vm.Name, nodeID, solutionType, nil, network.Name, disks, nil, []workloads.VM{vm}, nil)
+			deployment := workloads.NewDeployment(vm.Name, nodeID, solutionType, nil, network.Name, disks, nil, []workloads.VM{vm}, nil, nil)
```
Comment (rawdaGastan) [inline tfrobot/pkg/deployer/deployer.go:245]: I think we should open an issue for supporting volumes in tfrobot

File: grid-cli/internal/cmd/deploy.go [hunk 4]
```diff
@@ -27,7 +27,7 @@ func DeployVM(ctx context.Context, t deployer.TFPluginClient, vm workloads.VM, m
 		mounts = append(mounts, mount)
 	}
 	vm.NetworkName = networkName
-	dl := workloads.NewDeployment(vm.Name, node, projectName, nil, networkName, mounts, nil, []workloads.VM{vm}, nil)
+	dl := workloads.NewDeployment(vm.Name, node, projectName, nil, networkName, mounts, nil, []workloads.VM{vm}, nil, nil)
```
Comment (rawdaGastan) [inline grid-cli/internal/cmd/deploy.go:30]: I think we should open an issue for supporting volumes in `grid-cli`

File: grid-client/integration_tests/vm_volume_test.go [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+		t.Skipf("plugin creation failed: %v", err)
+	}
+
+	publicKey, privateKey, err := GenerateSSHKeyPair()
+	require.NoError(t, err)
+
+	nodes, err := deployer.FilterNodes(
+		context.Background(),
+		tfPluginClient,
+		generateNodeFilter(WithFreeSRU(3)),
```
Comment (rawdaGastan) [inline grid-client/integration_tests/vm_volume_test.go:26]: Why sru is 3 and you request 2 only in the next line?

**Review by rawdaGastan — COMMENTED** [review]
Great !! 🔥

File: grid-client/integration_tests/vm_volume_test.go (hunk 5 above)
Comment (AbdelrahmanElawady) [reply grid-client/integration_tests/vm_volume_test.go:26]: I don't really remember but it seems 1 for Volume and 2 for minimum rootfs, the 2 next line should be 1

**Review by rawdaGastan — APPROVED** [review]

**Pull request merged**